sdks/python/apache_beam/io/textio.py

```diff
@@ -175,6 +182,33 @@ def split_points_unclaimed(stop_position):
         if num_bytes_to_next_record < 0:
           break
 
+  def _process_header(self, file_to_read, read_buffer):
+    # Returns a tuple containing the position in file after processing header
+    # records and a list of decoded header lines that match
+    # 'header_matcher_predicate'.
+    header_lines = []
+    position = self._skip_lines(
+        file_to_read, read_buffer,
+        self._skip_header_lines) if self._skip_header_lines else 0
+    while self._header_matcher_predicate:
+      record, num_bytes_to_next_record = self._read_record(file_to_read,
+                                                           read_buffer)
+      decoded_line = self._coder.decode(record)
+      if not self._header_matcher_predicate(decoded_line):
+        # We've read past the header section at this point, so go back a line.
+        file_to_read.seek(position)
+        read_buffer.reset()
+        break
+      header_lines.append(decoded_line)
+      if num_bytes_to_next_record < 0:
+        break
+      position += num_bytes_to_next_record
+
+    if self._header_processor:
+      self._header_processor(header_lines)
```

Comment:
<!--new_thread; commit:70ade9d4fe6a8e31ff0be4d39b731510c93b4bf9; resolved:0-->
Move this to inside if block suggested above.